packages/plugin-svelte/src/index.ts

```diff
@@ -2,14 +2,40 @@ import path from 'path';
 import { logger } from '@rsbuild/core';
 import { deepmerge } from '@rsbuild/shared';
 import type { RsbuildPlugin } from '@rsbuild/core';
+import type { CompileOptions } from 'svelte/compiler';
+import type SveltePreprocess from 'svelte-preprocess';
+
+export interface SvelteLoaderOptions {
+  compilerOptions?: Omit<CompileOptions, 'filename' | 'format' | 'generate'>;
+  /**
+   * Extra HMR options, the defaults are completely fine\
+   * You can safely omit hotOptions altogether
+   */
+  hotOptions?: {
+    /** Preserve local component state */
+    preserveLocalState?: boolean;
+    [key: string]: any;
+  };
+  [key: string]: any;
+}
+
+export type AutoPreprocessOptions = NonNullable<
+  Parameters<typeof SveltePreprocess>[0]
+>;
 
 export type PluginSvelteOptions = {
   /**
    * The options of svelte-loader
    *
    * See https://github.com/sveltejs/svelte-loader
    */
-  svelteLoaderOptions?: Record<string, any>;
+  svelteLoaderOptions?: SvelteLoaderOptions;
+  /**
+   * The options of svelte-preprocess
+   *
+   * See https://github.com/sveltejs/svelte-preprocess
+   */
+  preprocessOptions?: AutoPreprocessOptions;
```

Comment:
Thanks! Can you add an unit test for this new option?

see: packages/plugin-svelte/tests/index.test.ts

Reply:
done :)

but the snapshot was not serialized expectably

Reply:
haha, maybe we can improve the snapshot in the future